File: PRJ321(Java-Web-Base)/Cart_JSTL/src/java/danhnlc/daos/ProductDAO.java
package danhnlc.daos;

import danhnlc.common.AbstractDAO;
import danhnlc.dtos.Product;
import danhnlc.util.DBConnection;
import java.util.ArrayList;
import java.util.List;

public class ProductDAO extends AbstractDAO {

    public List<Product> getAll() throws Exception {
        List<Product> result = null;
        String productID, productName;
        float price;
        int quantity;
        String sql = "Select productID, productName, price, quantity From tblProducts";
        try {
            conn = DBConnection.getConnection();
            preStm = conn.prepareStatement(sql);
            rs = preStm.executeQuery();
            result = new ArrayList<>();
            while (rs.next()) {
                productID = rs.getString("productID");
                productName = rs.getString("productName");
                price = rs.getFloat("price");
                quantity = rs.getInt("quantity");
                result.add(new Product(productID, productName, price, quantity));
            }
        } catch (Exception e) {
        } finally {
            closeConnection();
        }
        return result;
    }

    public boolean create(Product product) throws Exception {
        String sql = "Insert Into tblProducts(productID, productName, price, quantity) Values(?,?,?,?)";
        return query(sql, product.getProductID(), product.getProductName(), product.getPrice(), product.getQuantity());
    }

    public boolean update(Product product) throws Exception {
        String sql = "Update tblProducts Set productName=?, price=?, quantity=? Where productID=?";
        return query(sql, product.getProductName(), product.getPrice(), product.getQuantity(), product.getProductID());
    }

    public boolean delete(String productID) throws Exception {
        String sql = "Delete From tblProducts Where productID=?";
        return query(sql, productID);
    }
}

File: C#/ASS3/ProductLibrary/ProductLibrary/DAL/ProductDAO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Data;
using System.Data.SqlClient;
using ProductLibrary.DB;

namespace ProductLibrary.DAL
{
    class ProductDAO
    {
        public DataSet GetAll()
        {
            DataSet result = null;
            String sql = "Select ProductID, ProductName, UnitPrice, Quantity, (UnitPrice*Quantity) as SubTotal From Products";
            SqlConnection conn = DBConnection.GetConnection();
            SqlDataAdapter da = new SqlDataAdapter(sql, conn);
            conn.Open();
            DataSet dt = new DataSet();
            da.Fill(dt, "Products");
            result = dt;
            conn.Close();
            return result;
        }

        public DataSet FindByName(string name)
        {
            DataSet result = null;
            String sql = "Select ProductID, ProductName, UnitPrice, Quantity From Products Where ProductName Like @Name";
            SqlConnection conn = DBConnection.GetConnection();
            SqlCommand cmd = new SqlCommand(sql, conn);
            cmd.Parameters.AddWithValue("@Name", "%" + name + "%");
            SqlDataAdapter da = new SqlDataAdapter(cmd);
            conn.Open();
            DataSet dt = new DataSet();
            da.Fill(dt, "Products");
            result = dt;
            conn.Close();
            return result;
        }


        public bool Insert(Product product)
        {
            bool check = false;
            String sql = "Insert Products(ProductName, UnitPrice, Quantity) Values(@Name, @Price, @Quantity)";
            SqlConnection conn = DBConnection.GetConnection();
            conn.Open();
            SqlCommand cmd = new SqlCommand(sql, conn);
            cmd.Parameters.AddWithValue("@Name", product.ProductName);
            cmd.Parameters.AddWithValue("@Price", product.UnitPrice);
            cmd.Parameters.AddWithValue("@Quantity", product.Quantity);
            check = cmd.ExecuteNonQuery() > 0;
            conn.Close();
            return check;
        }

        public bool Update(Product product)
        {
            bool check = false;
            String sql = "Update Products Set ProductName=@Name, UnitPrice=@Price, Quantity=@Quantity Where ProductID=@ID";
            SqlConnection conn = DBConnection.GetConnection();
            conn.Open();
            SqlCommand cmd = new SqlCommand(sql, conn);
            cmd.Parameters.AddWithValue("@ID", product.ProductID);
            cmd.Parameters.AddWithValue("@Name", product.ProductName);
            cmd.Parameters.AddWithValue("@Price", product.UnitPrice);
            cmd.Parameters.AddWithValue("@Quantity", product.Quantity);
            check = cmd.ExecuteNonQuery() > 0;
            conn.Close();
            return check;
        }

        public bool Delete(int id)
        {
            bool check = false;
            String sql = "Delete Products Where ProductID=@ID";
            SqlConnection conn = DBConnection.GetConnection();
            conn.Open();
            SqlCommand cmd = new SqlCommand(sql, conn);
            cmd.Parameters.AddWithValue("@ID", id);
            check = cmd.ExecuteNonQuery() > 0;
            conn.Close();
            return check;
        }
    }
}
